proxy/envoy/resources.go

```diff
@@ -105,15 +121,20 @@ type Runtime struct {
 
 // Route definition
 type Route struct {
-	Runtime          *Runtime         `json:"runtime,omitempty"`
-	Path             string           `json:"path,omitempty"`
-	Prefix           string           `json:"prefix,omitempty"`
-	PrefixRewrite    string           `json:"prefix_rewrite,omitempty"`
+	Runtime       *Runtime `json:"runtime,omitempty"`
+	Path          string   `json:"path,omitempty"`
+	Prefix        string   `json:"prefix,omitempty"`
+	PrefixRewrite string   `json:"prefix_rewrite,omitempty"`
+
 	Cluster          string           `json:"cluster"`
 	WeightedClusters *WeightedCluster `json:"weighted_clusters,omitempty"`
-	Headers          []Header         `json:"headers,omitempty"`
-	TimeoutMS        int              `json:"timeout_ms,omitempty"`
-	RetryPolicy      *RetryPolicy     `json:"retry_policy,omitempty"`
+
+	Headers     Headers      `json:"headers,omitempty"`
+	TimeoutMS   int          `json:"timeout_ms,omitempty"`
+	RetryPolicy *RetryPolicy `json:"retry_policy,omitempty"`
+
+	// Special field to collect clusters
+	Clusters []Cluster `json:"-"`
```

Comment:
To create an AST of some sorts by inserting data into routes themselves.
These routes get passed around a lot, and we'd have to pass a pair of route, cluster, otherwise.